modules/auxiliary/gather/behind_cloudflare.rb

```diff
@@ -0,0 +1,445 @@
+  def search(keyword, search_type, uid, secret)
+    begin
+      payload  = {
+        'query' => keyword
+      }
+
+      cli      = Rex::Proto::Http::Client.new('www.censys.io', 443, {}, true)
+      cli.connect
+
+      response = cli.request_cgi(
+        'method'  => 'post',
+        'uri'     => "/api/v1/search/#{search_type}",
+        'headers' => {
+          'Authorization' => basic_auth_header(uid, secret)
+        },
+        'data'    => payload.to_json
+      )
+      results  = cli.send_recv(response)
+
+    rescue ::Rex::ConnectionError, Errno::ECONNREFUSED, Errno::ETIMEDOUT
+      print_error("HTTP Connection Failed")
+    end
+
+    unless results
+      print_error('server_response_error')
+      return false
+    end
+
+    records = ActiveSupport::JSON.decode(results.body)
+    results = records['results']
+
+    return parse_ipv4(results)
+  end
+
+  ## auxiliary/gather/censys_search.rb
+  def parse_ipv4(records)
+    ip_list = []
+    records.each do | ipv4 |
+      ip_list.push(ipv4['ip'])
+    end
+    return ip_list
+  end
+
+  ## auxiliary/gather/censys_search.rb
+  def valid_domain?(domain)
+    domain =~ /^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$/
+  end
+
+  def run
+    print_status('Passive gathering information...')
+
+    domain_name   = PublicSuffix.parse(datastore['HOSTNAME']).domain
+    ip_list       = []
+
+    # PrePost SEO
+    ip_records    = do_grab_domain_ip_history(domain_name, datastore['Proxies'])
+    ip_list      |= ip_records unless ip_records.eql? false
+    unless ip_records.eql? false
+      print_status(" * ViewDNS.info: #{ip_records.count.to_s} IP address found(s).")
+    end
+
+    # DNS Enum.
+    if datastore['DNSENUM'].eql? true
+      ip_records   = do_dns_enumeration(domain_name, datastore['THREADS'])
+      ip_list     |= ip_records unless ip_records.eql? false
+      unless ip_records.eql? false
+        print_status(" * DNS Enumeration: #{ip_records.count.to_s} IP address found(s).")
+      end
+    end
+
+    # Censys search.
+    if [datastore['CENSYS_UID'], datastore['CENSYS_SECRET']].none?(&:nil?)
+      ip_records  = search(domain_name, 'ipv4', datastore['CENSYS_UID'], datastore['CENSYS_SECRET'])
+      ip_list    |= ip_records unless ip_records.eql? false
+      unless ip_records.eql? false
+        print_status(" * Censys IPv4: #{ip_records.count.to_s} IP address found(s).")
+        print_status()
+      end
+    end
+
+    unless ip_list.empty?
+
+      # Cleaning the results.
+      print_status("Clean cloudflare server(s)...")
+      ip_blacklist = get_cloudflare_ips
+      records      = []
+      ip_list.uniq.each do | ip |
+        is_listed = false
+
+        ip_blacklist.each do | ip_range |
+          if IPAddr.new(ip_range).include? ip.to_s
+            is_listed = true
+            break
+          end
+        end
+
+        unless is_listed.eql? true
+          records << ip.to_s
+        end
+      end
+
+      if records.empty?
```

Comment:
Same here. If there's no records, there's no point continuing. Also, `records.count.to_s` should always be zero here.

```ruby
if records.empty?
  print_bad(' * TOTAL: No IP address found after cleaning.')
  return
end
```